tephra-core/src/test/java/co/cask/tephra/persist/LocalTransactionStateStorageTest.java

```diff
@@ -49,4 +66,99 @@ protected Configuration getConfiguration(String testName) throws IOException {
+      }
+    };
+
+    TransactionStateStorage storage = null;
+    try {
+      storage = getStorage(conf);
+      storage.startAndWait();
+
+      // Create transaction snapshot and transaction edits with version when long running txns had -1 expiration.
+      Collection<Long> invalid = Lists.newArrayList();
+      NavigableMap<Long, TransactionManager.InProgressTx> inProgress = Maps.newTreeMap();
+      long time1 = System.currentTimeMillis();
+      inProgress.put(time1, new TransactionManager.InProgressTx(time1 - 5, -1));
+      long time2 = time1 + 100;
+      inProgress.put(time2, new TransactionManager.InProgressTx(time2 - 50, time2 + 1000));
+      Map<Long, Set<ChangeId>> committing = Maps.newHashMap();
+      Map<Long, Set<ChangeId>> committed = Maps.newHashMap();
+      TransactionSnapshot snapshot = new TransactionSnapshot(time1, 0, time1, invalid,
+                                                             inProgress, committing, committed);
+      long time3 = time1 + 200;
+      TransactionEdit edit1 = new TransactionEditV2(time3, time3 - 10, TransactionEdit.State.INPROGRESS, -1,
+                                                    null, 0L, false, null);
+      long time4 = time1 + 300;
+      TransactionEdit edit2 = new TransactionEditV2(time4, time4 - 10, TransactionEdit.State.INPROGRESS, time4 + 1000,
+                                                    null, 0L, false, null);
+
+      // write snapshot and transaction edit
+      storage.writeSnapshot(snapshot);
+      TransactionLog log = storage.createLog(time1);
+      log.append(edit1);
+      log.append(edit2);
+      log.close();
+
+      // Start transaction manager
+      conf.setInt(TxConstants.Manager.CFG_TX_CLEANUP_INTERVAL, 3);
+      conf.setInt(TxConstants.Manager.CFG_TX_TIMEOUT, 2);
+      conf.setInt(TxConstants.Manager.CFG_TX_LONG_TIMEOUT, 2);
+      conf.set(TxConstants.Persist.CFG_TX_SNAPHOT_CODEC_CLASSES, latestSnapshotCodec);
+      TransactionManager txm = new TransactionManager(conf, storage, new TxMetricsCollector());
+      txm.startAndWait();
+      try {
+        //Verify that the txns in old format were read correctly.
+        //There should be four in-progress transactions, and no invalid transactions
+        TransactionSnapshot snapshot1 = txm.getCurrentState();
+        Assert.assertEquals(ImmutableSortedSet.of(time1, time2, time3, time4), snapshot1.getInProgress().keySet());
+        verifyInProgress(snapshot1.getInProgress().get(time1), TransactionType.LONG, time1);
+        verifyInProgress(snapshot1.getInProgress().get(time2), TransactionType.SHORT, time2);
+        verifyInProgress(snapshot1.getInProgress().get(time3), TransactionType.LONG, time3);
+        verifyInProgress(snapshot1.getInProgress().get(time4), TransactionType.SHORT, time3);
+        Assert.assertEquals(0, snapshot1.getInvalid().size());
+
+        // sleep longer than the cleanup interval, so that txns get timed out
+        TimeUnit.SECONDS.sleep(5);
```

Comment:
You are right, just verifying the following should be enough:
* Long running txn - expiry time >= TransactionManager start time + long txn timeout
* Short txn - expiry time == expiry time given during creation

I will remove the wait for timeout to happen.